src/uwtools/tests/test_schemas.py

```diff
@@ -90,6 +96,117 @@ def test_schema_chgres_cube_run_dir(chgres_cube_prop):
     assert "88 is not of type 'string'" in errors(88)
 
 
+# esg-grid
+
+
+def test_schema_esg_grid():
+    config = {
+        "execution": {"executable": "esg_grid"},
+        "run_dir": "/tmp",
+    }
+    errors = schema_validator("esg-grid", "properties", "esg_grid")
+    # Basic correctness:
+    assert not errors(config)
+    # Some top-level keys are required:
+    for key in ("execution", "run_dir"):
+        assert f"'{key}' is a required property" in errors(with_del(config, key))
+    # Additional top-level keys are not allowed:
+    assert "Additional properties are not allowed" in errors({**config, "foo": "bar"})
+    # Additional MPI support is not allowed:
+    assert "Additional properties are not allowed ('mpicmd' was unexpected)" in errors(
+        {"execution": {"mpicmd": "srun"}}
+    )
+
+
+def test_schema_esg_grid_namelist(esg_grid_prop):
+    base_file = {"base_file": "/some/path"}
+    update_values = {
+        "update_values": {
+            "regional_grid_nml": {
+                "delx": 0.22,
+                "dely": 0.22,
+                "lx": -200,
+                "ly": -130,
+                "pazi": 0.0,
+                "plat": 45.5,
+                "plon": -100.5,
+            }
+        }
+    }
+    config = """
+        "regional_grid_nml": {
+            "delx": 0.22,
+            "dely": 0.22,
+            "lx": -200,
+            "ly": -130,
+            "pazi": 0.0,
+            "plat": 45.5,
+            "plon": -100.5,
+        }
+    """
+    cfg = yaml.safe_load(config)
+
+    errors = esg_grid_prop("namelist")
+    # Just base_file is ok:
+    assert not errors(base_file)
+    # base_file must be a string:
+    assert "88 is not of type 'string'" in errors({"base_file": 88})
+    # Just update_values is ok:
+    assert not errors(update_values)
+    # A combination of base_file and update_values is ok:
+    assert not errors({**base_file, **update_values})
+    # All key/value pairs in update_values must be present if no base_file is supplied:
+    assert "'dely' is a required property" in errors(
+        {"update_values": {"regional_grid_nml": {"delx": 0.11, "lx": -180, "plat": 38.0}}}
+    )
+
+    # Subsection of update_values is ok if base_file is supplied:
+    #  assert not errors(
+    #      {
+    #          **base_file,
+    #          "update_values": {"regional_grid_nml": {"delx": 0.11, "lx": -180, "plat": 38.0}},
+    #      }
+    #  )
```

Comment:
I thought providing `base_file` and a subsection of `update_values` would pass the schema, but it's currently failing unless I supply all the key/value pairs for `update_values`. I thought this assertion should pass the schema, but now i'm starting to second guess if it actually correctly presenting an error. 

Thoughts?

Reply:
I think the schema might not be constructed properly: The single item in the `allOf` list at the very top has an `if` item that refers to something called `namelist.base_file`, but I wouldn't think, at that level of the JSON structure, that would refer to what we'd hope: The `base_file` block, from this top level, is found by following the keys `properties` -> `esg_grid` -> `properties` -> `namelist` -> `properties` -> `base_file`. Maybe JSON Schema understand a simplified version of that that elides the `properties` steps/keys, but at the very least I assume the `esg_grid` level would have to be taken into account for a relative reference. I don't think I've done anything in JSON Schema with anything but immediate-neighbor properties, so this is a gray area for me.

Also, the `then` item of that block just refers to `namelist`, and I can't see how JSON Schema could connect that to the more deeply nested `namelist` block that we want to talk about.

I feel like the definition of which kind of `update_values` block is expected should be happening deeper in the schema, closer to the `base_file` and `update_values` blocks themselves.